internals/secrethub/service_aws_init.go

```diff
@@ -97,3 +98,22 @@ func (cmd *ServiceAWSInitCommand) Register(r Registerer) {
 
 	BindAction(clause, cmd.Run)
 }
+
+// roleNameFromRole returns the name of the role indicated by the input. Accepted input is:
+// - A role name (e.g. my-role)
+// - A role name, prefixed by "role/" (e.g. role/my-role)
+// - A role ARN (e.g. arn:aws:iam::123456789012:role/my-role)
+//
+// When the input is not one of these accepted inputs, no guarantees about the expected return
+// are made.
+func roleNameFromRole(role string) string {
+	if strings.Contains(role, ":") {
+		parts := strings.SplitN(role, "role/", 2)
+		if len(parts) == 2 {
+			return parts[1]
+		}
+	} else {
+		return strings.TrimPrefix(role, "role/")
+	}
+	return ""
```

Comment:
Can we rewrite this code to be a bit easier to evaluate? E.g. moving this return to just under the `if len(parts) == 2` check on line 114 and removing the else statement on line 115? Took me a few WTFs to understand that this line is actually not dead code.